src/system/group.cr

```diff
@@ -0,0 +1,195 @@
+    !LibC.getgrgid(int_to_gid(gid)).null?
+  end
+
+  # Checks to see if group with a given groupname exists.
+  #
+  # Returns: `true` if the group exists.
+  #
+  # ```
+  # System::Group.exists?(0)     # => true
+  # System::Group.exists?(32766) # => false
+  # ```
+  def self.exists?(group : String) : Bool
+    !LibC.getgrnam(group.check_no_null_byte).null?
+  end
+
+  # Returns the group specified by the group ID.
+  # ```
+  # System::Group.get(0)
+  # ```
+  def self.get(gid : Int) : Group
+    new(gid)
+  end
+
+  # Returns the group specified by the groupname.
+  # ```
+  # System::Group.get("root")
+  # ```
+  def self.get(groupname : String) : Group
+    new(groupname)
+  end
+
+  # Initializes a group by the given group ID.
+  def initialize(gid : Int)
+    group_struct = LibC.getgrgid(Group.int_to_gid(gid))
+    raise NotFound.new("Group with gid '#{gid}', was not found.") if group_struct.null?
+
+    group_struct = group_struct.value
+    @name = String.new(group_struct.gr_name)
+    @gid = group_struct.gr_gid
+    slice = group_struct.gr_mem.to_slice_null_terminated(LibC::NGROUPS_MAX)
+    @member_names = Array.new(slice.size) { |idx| String.new(slice[idx]) }
+  end
+
+  # Initializes a group by the given groupname.
+  def initialize(groupname : String)
+    group_struct = LibC.getgrnam(groupname.check_no_null_byte)
+    raise NotFound.new("Group with name '#{groupname}', was not found.") if group_struct.null?
+
+    group_struct = group_struct.value
+    @name = String.new(group_struct.gr_name)
+    @gid = group_struct.gr_gid
+    slice = group_struct.gr_mem.to_slice_null_terminated(LibC::NGROUPS_MAX)
+    @member_names = Array.new(slice.size) { |idx| String.new(slice[idx]) }
+  end
+
+  # Returns the groups name.
+  #
+  # ```
+  # System::Group.get(0).name # => "root"
+  # ```
+  getter name : String
+
+  # Returns the group ID.
+  #
+  # ```
+  # System::Group.get("root").gid # => 0
+  # ```
+  getter gid : UInt32
+
+  # Returns an array of the groups members names.
+  #
+  # ```
+  # System::Group.get("root").member_names
+  # ```
+  getter member_names : Array(String)
+
+  # Yields the groups members names.
+  #
+  # ```
+  # System::Group.get("root").member_names() { |name| puts names }
+  # ```
+  def each_member_name(&block : String ->) : Nil
+    @member_names.each { |member| yield(member) }
+  end
+
+  # Returns an array of the groups members.
+  #
+  # ```
+  # System::Group.get("root").members
+  # ```
+  def members : Array(User)
+    @member_names.map { |member| User.get(member) }
+  end
+
+  # Yields the groups members.
+  #
+  # ```
+  # System::Group.get("root").members() { |user| puts user.name }
+  # ```
+  def each_member(&block : User ->) : Nil
+    each_member_name { |member| yield(User.get(member)) }
+  end
+
+  # See `Object#hash(hasher)`
+  def hash(hasher)
+    @gid.hash(hasher)
+  end
+
+  # Returns `true` if `self` is equal to *other*.
+  def ==(other : Group) : Bool
```

Comment:
This definition is redundant since there's already defined `def <=>(other : Group)`.

Reply:
Fixed.